### PR TITLE
⚡️ Improved handling of Falsy parameters in hooks

### PR DESCRIPTION
**Pull request by PoulJohny**
I've found that most of the hooks support passing Falsy as parameters but not all, so I've:
- added possibility to pass Falsy as a contract in useContractFunction
- fixed throwing error when passed Falsy as useToken address parameter

### CODE REVIEW

**Review by dmaretskyi — APPROVED** [review]

File: packages/core/src/hooks/useToken.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ import { Contract } from 'ethers'
  * @public
  */
 export function useToken(tokenAddress: string | Falsy): TokenInfo | undefined {
+  if (!tokenAddress) {
```
Comment (dmaretskyi) [inline packages/core/src/hooks/useToken.ts:11]: I don't think you can do that. Every call to the hook must invoke the same set of hooks in the same order every time. This one misses the `useCalls` below.

https://reactjs.org/docs/hooks-rules.html

**Review by dmaretskyi — CHANGES_REQUESTED** [review]

File: packages/core/src/hooks/useToken.ts (hunk 1 above)
Comment (noobshow) [reply packages/core/src/hooks/useToken.ts:11]: I think the result might be undefined regardless of this check.
Didn't test it though.

Comment (PoulJohny) [reply packages/core/src/hooks/useToken.ts]: @dmaretskyi you are right, knew about it but forgot to change. Now I've pushed updated code. Please check :)

**Review by mj426382 — APPROVED** [review]

**Comment by gpxl-dev** [comment]
@dmaretskyi - looks like this PR Is blocked by an outdated change request?
I've hit an issue caused by the error thrown in `useToken` with falsy params and would love to see this merged.

**Review by rzadp — APPROVED** [review]

**Pull request merged**